src/Main.elm

```diff
@@ -1,10 +1,829 @@
+                        |> Dict.foldl viewFolder []
+
+        filteredAudiences : List (E.Element Msg)
+        filteredAudiences =
+            case model.sortBy of
+                Data.Audience.Shared ->
+                    model.audiences
+                        |> Dict.filter (\key value -> value.type_ == model.sortBy)
+                        |> Dict.foldl viewAudience []
+
+                _ ->
+                    model.audiences
+                        |> Dict.filter (\key value -> value.folder == currentFolder)
+                        |> Dict.filter (\key value -> value.type_ == model.sortBy)
+                        |> Dict.foldl viewAudience []
+
+        -- we would normally use an icon here that would represent "i" for
+        -- "info", but a regular text element did the job well enough
+        coolI : E.Element Msg
+        coolI =
+            E.el
+                [ E.width <| E.px 15
+                , E.height <| E.px 15
+                , EFont.color Palette.color2
+                ]
+                <| E.el [ E.centerX, E.centerY ] <| E.text "i"
+
+        -- according to "FIRST_STEP.md", we should only display the "back" button
+        -- if we're not at the root of the tree
+        upperLeftBackButtonOrText : E.Element Msg
+        upperLeftBackButtonOrText =
+            case currentFolder of
+                Just _ ->
+                    EInput.button
+                        [ E.centerY
+                        , EBackground.color Palette.color4
+                        , EBorder.rounded 3
+                        , E.paddingXY 7 7
+                        ]
+                        { onPress = Just GoBack
+                        , label = E.text "Go Back"
+                        }
+
+                Nothing ->
+                    E.el
+                        [ E.centerY
+                        ]
+                        <| E.text upperLeftText
+
+        upperLeftText : String
+        upperLeftText =
+            case model.sortBy of
+                Data.Audience.Shared ->
+                    "Displaying Shared audiences from all folders"
+
+                _ ->
+                    "Click a folder to start browsing!"
+
+        upperRightInfoElement : E.Element Msg
+        upperRightInfoElement =
+            E.row
+                [ EBackground.color Palette.color4
+                , E.padding 5
+                , E.spacing 5
+                , EBorder.roundEach { topRight = 99, bottomRight = 99, bottomLeft = 20, topLeft = 20 }
+                , EFont.color Palette.gray0
+                ]
+                [ E.paragraph
+                    [ E.paddingXY 5 0
+                    ]
+                    [ E.text "Currently sorting by "
+                    , E.el
+                        [ EFont.bold ]
+                        <| E.text (Data.Audience.audienceTypeToString model.sortBy)
+                    ]
+                , wrapColorCircle Palette.color3 coolI -- we wrap it in a colored circle
```

Comment:
Sorry and no offence, I just had to :D 
![thank-you-captain-obvious](https://user-images.githubusercontent.com/5626634/95427441-df61f900-0947-11eb-94b3-0c8982956a9b.jpg)

Reply:
hahaaaa now that you mention it, it's super obvious. )))))